### DISCUSSION TITLE
problem in intersection with structured grid

### DISCUSSION THREAD

**Discussion by v-alisei** [pyvista/pyvista#3432] [Q&A] [answered]
```
from math import pi
import pyvista as pv
import numpy as np


#forming a sphere of points
length = 2000
R = 3
alpha = np.random.uniform(0, pi*2, length)
phi = np.square(np.random.uniform(0, np.sqrt(pi/2), length))
x = np.cos(alpha) * np.cos(phi) * R 
y = np.sin(alpha) * np.cos(phi) * R
z = np.sin(phi) * R

# forming a cutting surface
lx = np.linspace(-5,5,100)
lz = np.linspace(-5,5,100)
lx, lz = np.meshgrid(lx, lz)
ly = np.sin(lx*3) + 1.5
cutsurf = pv.StructuredGrid(lx, ly, lz)

# reconstructiong a sphere
cords = np.transpose([x, y, z])
points = pv.PointSet(cords).cast_to_polydata()
surf = points.reconstruct_surface(sample_spacing=0.05)
print(surf.area)


cutline, s1_split, s2_split = surf.intersection(cutsurf.extract_surface())

# plotting
p1 = pv.Plotter()
p1.add_mesh(surf)
p1.add_mesh(points)
p1.add_mesh(cutsurf)
# p1.add_mesh(cutline)
p1.show()
```
yields

`58.48021245682933
2022-10-08 18:47:28.012 (   1.835s) [                ]    vtkPointLocator.cxx:868    ERR| vtkPointLocator (000001AB9CF53B10): No points to subdivide  
ERROR:root:No points to subdivide
2022-10-08 18:47:28.013 (   1.837s) [                ]vtkIntersectionPolyData:2410  WARN| No Intersection between objects
`

Hello, everyone. I've ran into problem. I need to get a line of intersection between a surface (surf) and a structuredGrid (cutsurf). Structured grid can't be used in intersection. For now I've tried to use `cutsurf.extract_surface()` , `cutsurf.extract_geometry()` and many other things. Basically, I need to "put down" an arbitrary line in xy plane onto any surface, so I decided to extend the line in z direction making it a cutting surface, and then find an intersection. So maybe I should generate polyData myself somehow? Can you help me please?
![image](https://user-images.githubusercontent.com/93995895/194716503-a2f1bc7f-9d50-4efa-845c-dda0017042c5.png)

**Comment by darikg** [comment, marked as answer]
I think `cutsurf.extract_surface().triangulate()` does it

**Reply by v-alisei** [reply]
Omg thank you! You were really helphul)
